src/fieldAnalyser/fieldAnalyser.ts

```diff
@@ -0,0 +1,138 @@
+import PlatformClient, {FieldTypes} from '@coveord/platform-client';
+import {BatchUpdateDocuments, DocumentBuilder} from '..';
+
+type fieldTypeDict = {[key in FieldTypes]?: number};
+type typeOccurence = [fieldType: string, occurence: number];
+export class FieldAnalyser {
```

Comment:
Early optimization is the root of all evil, but we should do a sanity check with a realistic scenario with lots of documents for this class.

Since we store all existing, missing, and inconsistent fields in memory, this might be too much ?

Reply:
I agree and these data structures "should" be very light in memory.
Assuming that the field limit is set to 5000 and that a field cannot be more than 255 characters, These data structures will not store more than ~2MB of data.

With that said, I'll still perform some more performance/memory analysis when I will "plug" the `FieldAnalyser` module into the `Source` class.

Also, I still need to do some optimization to prevent `JavaScript heap out of memory` errors (which will be part of the CDX-840).